tracing-attributes/src/attr.rs

```diff
@@ -199,29 +204,29 @@ impl Parse for Skips {
 }
 
 #[derive(Clone, Debug, Hash, PartialEq, Eq)]
-pub(crate) enum ErrorMode {
+pub(crate) enum FormatMode {
     Display,
     Debug,
 }
 
-impl Default for ErrorMode {
+impl Default for FormatMode {
     fn default() -> Self {
-        ErrorMode::Display
+        FormatMode::Display
     }
 }
 
-impl Parse for ErrorMode {
+impl Parse for FormatMode {
     fn parse(input: ParseStream<'_>) -> syn::Result<Self> {
         if !input.peek(syn::token::Paren) {
-            return Ok(ErrorMode::default());
+            return Ok(FormatMode::default());
         }
         let content;
         let _ = syn::parenthesized!(content in input);
         let maybe_mode: Option<Ident> = content.parse()?;
-        maybe_mode.map_or(Ok(ErrorMode::default()), |ident| {
+        maybe_mode.map_or(Ok(FormatMode::default()), |ident| {
```

Comment:
I don't think `Display` should be the default for return values, personally. Most return values will likely not implement `Display` --- `Display` was the default for errors simply because most errors implement the `std::error::Error` trait, which requires them to be `Display`. If we're always going to format return values, we should probably default to using `Debug`, and make `Display` be the one that the user opts in to.

With that said, I think we may actually want to change so that the default behavior is to record the return value as a `tracing` primitive value if it's one of the primitive `Value` types, and fall back to `Display` if it's not. We can look at the function signature's return type, and we already have code for determining whether a type is a primitive `Value`: https://github.com/tokio-rs/tracing/blob/f513f4aaa2bd3fef19e28eaa0532b672fd1cb976/tracing-attributes/src/expand.rs#L345-L398

Reply:
Because we are special-casing `Result<T, E>`s, in order to do fallback'ing we need to extract `T` from the `Result` of `syn::Type` being returned, which I can find the way to do it. Therefore I'd go with the former change.

Reply:
Hmm, that's fair; I think it _should_ be possible to find `Result`s and get the type parameter using `syn`, but it might be fairly complex. We can always add that in a future change.